app/routes.js

```diff
@@ -1,7 +1,9 @@
 const { healthCheck } = require('./controllers/healthCheck');
 const userController = require('./controllers/users');
+const { userSchema } = require('./schemas/users');
+const { validateBySchema, existUserDB } = require('./middlewares/users');
 
 exports.init = app => {
   app.get('/health', healthCheck);
-  app.post('/users', userController.createUser);
+  app.post('/users', [existUserDB, validateBySchema(userSchema)], userController.createUser);
```

Comment:
It could be that you first evaluate the scheme, this way it makes sure that you will search with a valid email in the database.